sklearn/compose/tests/test_target.py

```diff
@@ -332,3 +332,22 @@ def test_transform_target_regressor_route_pipeline():
     pip.fit(X, y, **{'est__check_input': False})
 
     assert regr.transformer_.fit_counter == 1
+
+
+class DummyRegressorWithExtraPredictParams(DummyRegressor):
+    def predict(self, X, check_input=True):
+        # In the test below we make sure that the check input parameter is
+        # passed as false
+        assert not check_input
+        return super().predict(X)
+
+
+def test_transform_target_regressor_pass_extra_predict_parameters():
+    X, y = friedman
+    regr = TransformedTargetRegressor(
+        regressor=DummyRegressorWithExtraPredictParams(),
+        transformer=DummyTransformer()
+    )
+
+    regr.fit(X, y)
+    regr.predict(X, check_input=False)
```

Comment:
Thinking about it more, I think we can resolve this by using a simpler mock:

```py

from unittest.mock import Mock


def test_transform_target_regressor_pass_extra_predict_parameters():
    X, y = friedman
    regr = TransformedTargetRegressor(
        regressor=DummyRegressor(),
        transformer=DummyTransformer()
    )

    regr.fit(X, y)
    mock_predict = Mock(side_effect=regr.regressor_.predict)
    regr.regressor_.predict = mock_predict

    regr.predict(X, return_std=False)
    assert mock_predict.called
    assert mock_predict.call_args[1] == {"return_std": False}
```

Reply:
No DummyRegressorWithExtraPredictParams class needed, right? In this way, could the test for the fit method also be simplified using a Mock?

Reply:
> In this way, could the test for the fit method also be simplified using a Mock?

There may have been a way, but maybe mocking like this would be too hard to follow for `fit`.

The issue with the current test is that if `regr.predict` never calls `regressor_.predict`, the test will still pass. We need a way to make sure that `reg.regressor_.predict` is actually called to make sure `assert not check_input` ran. In the tests above there was a `fit_counter` that was checked in the test. We can have a `predict_called` as well:

```python
class DummyRegressorWithExtraPredictParams(DummyRegressor):
    def predict(self, X, check_input=True):
        # In the test below we make sure that the check input parameter is
        # passed as false
        self.predict_called = True
        assert not check_input
        return super().predict(X)

def test_transform_target_regressor_pass_extra_predict_parameters():
    # Checks that predict kwargs are passed to regressor.
    X, y = friedman
    regr = TransformedTargetRegressor(
        regressor=DummyRegressorWithExtraPredictParams(),
        transformer=DummyTransformer()
    )
    regr.fit(X, y)
    regr.predict(X, check_input=False)
    assert regr.regressor_.predict_called
```

Reply:
Ok, I will implement these changes. Thank you for the feedback!